src/module/actor/character/crafting/entry.ts

```diff
@@ -117,11 +114,13 @@ export class CraftingEntry implements CraftingEntryData {
             return false;
         }
 
-        if (!this.requiredTraits.some((traits) => traits.every((t) => formula.item.traits.has(t)))) {
+        const predicate = new PredicatePF2e(this.craftableItems ?? {});
```

Comment:
Wait, is this not already a PredicatePF2e?

Reply:
Not at this point, I should have it be a Predicate earlier, when it's passed to the constructor

Reply:
Actually no, since the CraftingEntry stores it's own data on a character, it needs to RawPredicate

Reply:
But it already is in the constructor. You have it as PredicatePF2e on line 33.

Also CraftingEntry is hopefully not persisted on the character, so it being an object should be fine. Please tell me its not persisted.

Reply:
It is persisted and has caused some headaches: that's a separate issue, unfortunately. For the RE, `Predicate.validate` can be used to check the raw predicate.

Reply:
It is currently persisted, to persist the formula on the entry

Reply:
So am I leaving this as is and then coming back after to do a non-persistence pass?

Reply:
Should be fine, though now I'm concerned that the constructor is doing new PredicatePF2e